feat(HTTP): Refresh 3XX response pages (#35353)
mdn/content: files/en-us/web/http/status/301/index.md

{{HTTPSidebar}}

The HTTP **`301 Moved Permanently`** [redirection response](/en-US/docs/Web/HTTP/Status#redirection_messages) status code indicates that the requested resource has been definitively moved to the URL in the {{HTTPHeader("Location")}} header.

> A browser redirects automatically to the new page, and the ranking value in search engines (colloquially named "link juice" in {{Glossary("SEO")}} terms) is sent to the new URL.

Reviewer: I would put this on a new line and clarify the search engine stuff (or remove it). I THINK the bit about search engines is wrong/ambiguous. My interpretation is below.

```suggestion

A browser receiving this status will automatically request the resource at the URL in the `Location` header, redirecting the user to the new page.
Search engines receiving this response will attribute any links to the original URL to the final resource, contributing to its SEO search ranking.
```

Author: I think it's important to mention the SEO impact of redirects, so we should keep it. Confirming if you redirect `page1 -> page2` with a 301, then page2 will have the ranking of the original (page1).

Author: I've made some edits, happy to see what you think / revise again

> [!NOTE]
> Although the [specification](#specifications) requires the request method and the body to remain unchanged when redirection is performed, not all user-agents meet this requirement.
> Use the `301` code only as a response for {{HTTPMethod("GET")}} or {{HTTPMethod("HEAD")}} methods and use the {{HTTPStatus("308", "308 Permanent Redirect")}} for other methods (typically {{HTTPMethod("POST")}}), as altering the method when handling `308` responses is explicitly prohibited.

## Status

```http
301 Moved Permanently
```

## Examples

### 301 response to a moved resource

The following {{HTTPMethod("GET")}} request to is made to a resource with a `301` redirection in place.

> The {{HTTPHeader("Location")}} header shows the new URL in place for the resource:

Reviewer: ```suggestion
The {{HTTPHeader("Location")}} header provides the URL where the resource has moved.
```

Reviewer: Suggestions don't work in this case, but my preference would be that we move this second line after the request. So essentially rather than say here's our request and response, we say First we make this request, then we get this response.

Author: Good call, update incoming

{{HTTPSidebar}}

The HTTP **`301 Moved Permanently`** [redirection response](/en-US/docs/Web/HTTP/Status#redirection_messages) status code indicates that the requested resource has been permanently moved to the URL in the {{HTTPHeader("Location")}} header.

A browser receiving this status will automatically request the resource at the URL in the `Location` header, redirecting the user to the new page.
Search engines receiving this response will attribute links to the original URL to the redirected resource, passing the {{Glossary("SEO")}} ranking to the new URL.

> [!NOTE]
> Although the [specification](#specifications) requires the request method and the body to remain unchanged when redirection is performed, not all user-agents meet this requirement.
> Use the `301` code only as a response for {{HTTPMethod("GET")}} or {{HTTPMethod("HEAD")}} methods and use the {{HTTPStatus("308", "308 Permanent Redirect")}} for other methods (typically {{HTTPMethod("POST")}}), as altering the method when handling `308` responses is explicitly prohibited.

## Status

```http
301 Moved Permanently
```

## Examples

### 301 response to a moved resource

> The following {{HTTPMethod("GET")}} request to is made to a resource with a `301` redirection in place.

Reviewer: ```suggestion
The following {{HTTPMethod("GET")}} request is made to a resource with a `301` redirection in place.
```
